### PR TITLE
Rename DisjointSet and repair constructor

### PR DESCRIPTION
**Pull request by c-p-murphy**
Finally got around to putting this together—sorry for the delay.

### CODE REVIEW

**Comment by c-p-murphy** [comment]
Also, I was on the fence about whether or not to hyphenate `Disjoint-Set`, but I ultimately decided to go with it ([Disjoint-set data structure](https://en.wikipedia.org/wiki/Disjoint-set_data_structure)). If people think that it would make more sense to have it as `Disjoint Set` though—just let me know.

File: src/disjoint_set.jl [hunk 1]
```diff
@@ -8,41 +8,41 @@
...
 """
-mutable struct IntDisjointSets{T<:Integer}
+mutable struct IntDisjointSet{T<:Integer}
     parents::Vector{T}
     ranks::Vector{T}
     ngroups::T
 
     # creates a disjoint set comprised of n singletons
-    # IntDisjointSets(n::T) where {T<:Integer} = new(collect(Base.OneTo(n)), zeros(T, n), n)
+    # IntDisjointSet(n::T) where {T<:Integer} = new(collect(Base.OneTo(n)), zeros(T, n), n)
```
Comment (oxinabox) [inline src/disjoint_set.jl:32]: Can we delete this line?

**Review by oxinabox — COMMENTED** [review]
I am going to need to review this carefuly to identify which lines changed and which were just renaming.

**Comment by c-p-murphy** [comment]
Hey @oxinabox, I think this should be good to go now—thanks for taking the time to review this.

**Pull request closed without merging**